packages/relay/src/lib/precheck.ts

```diff
@@ -116,11 +114,55 @@ export class Precheck {
       }
     } catch (error: any) {
       this.logger.trace('Error on balance precheck for sendRawTransaction(transaction=%s, totalValue=%s, error=%s)', transaction, txTotalValue, error.message);
-      
+
       result.passes = false;
       result.error = predefined.INTERNAL_ERROR;
     }
 
     return result;
   }
+
+  /**
+   * @param transaction
+   */
+  async gasLimit(transaction: string) {
+    const tx = ethers.utils.parseTransaction(transaction);
+    const gasLimit = tx.gasLimit.toNumber();
+    const failBaseLog = 'Failed gasLimit precheck for sendRawTransaction(transaction=%s).';
+
+    const intrinsicGasCost = Precheck.transactionIntrinsicGasCost(tx.data, tx.to);
+
+
+    if (gasLimit > constants.BLOCK_GAS_LIMIT) {
+      this.logger.trace(`${failBaseLog} Gas Limit was too high: %s, block gas limit: %s`, transaction, gasLimit, constants.BLOCK_GAS_LIMIT);
+      throw predefined.GAS_LIMIT_TOO_HIGH;
+    } else if (gasLimit < intrinsicGasCost) {
+      this.logger.trace(`${failBaseLog} Gas Limit was too low: %s, intrinsic gas cost: %s`, transaction, gasLimit, intrinsicGasCost);
+      throw predefined.GAS_LIMIT_TOO_LOW;
+    }
+  }
+
+  /**
+   * Calculates the intrinsic gas cost based on the number of empty bytes and whether the transaction is CONTRACT_CREATE
+   * @param data
+   * @param to
+   * @private
+   */
+  private static transactionIntrinsicGasCost(data: string, to: string | undefined) {
+    const isCreate = (to == undefined) || (to.length == 0);
+
+    let zeros = 0;
+
+    const dataBytes = Buffer.from(data, "hex");
+
+    for (const c of dataBytes) {
+      if (c == 0) {
+        zeros++;
+      }
+    }
+
+    const nonZeros = data.replace('0x', '').length - zeros;
```

Comment:
This was causing a wrong calculation of the gas cost. If `0x` is not removed from `data`, that counts as 2 `nonZero` bytes, and the calculated gas cost is `TX_BASE_COST + 2 * ISTANBUL_TX_DATA_NON_ZERO_COST`, which meant that transactions with empty data (hbar transfer transactions) could not pass the precheck if the default gasLimit (21000) was used.